ambry-store/src/main/java/com.github.ambry.store/IndexSegment.java

```diff
@@ -286,23 +298,26 @@ IndexValue find(StoreKey keyToFind) throws StoreException {
             if (result == 0) {
               byte[] buf = new byte[valueSize];
               duplicate.get(buf);
-              return new IndexValue(startOffset.getName(), ByteBuffer.wrap(buf));
+              toReturn = new IndexValue(startOffset.getName(), ByteBuffer.wrap(buf));
+              break;
             } else if (result < 0) {
               low = mid + 1;
             } else {
               high = mid - 1;
             }
           }
-          metrics.bloomFalsePositiveCount.inc(1);
+          if (toReturn == null && bloomPositive) {
```

Comment:
this is how it will look with your suggestion

```
if ( bloomFilter != null && !bloomFilter.isPresent(key)){
  // true negative
  if(bloomFilter != null){
     bloomAccessedCount.inc();
  }
} else{
   if(bloomFilter == null) {
      logger trace statement that bloom is null
   } else{
      logger trace
      bloomAccessedCount.inc()
      bloomPositiveCount.inc()
   }
  // binary search which will set toReturn
  // code block to do binary search in mmap
  if(bloomFilter != null && toReturn == null) {
     bloomFalsePositiveCount.inc()
  }
}